### ISSUE TITLE
ReStock: SM-6A Service Module

### ISSUE THREAD

**Issue by hockeygoalie35** [pizzaoverhead/FreeIva#404]
The Restock variant of the Making History 1.25 to 0.625m "SM-6A Service Module" has an option for a docking tunnel. Any chance this can be added to the next release? (ignore the naming in the photo)

![Snipaste-2024-03-08-19-30-25](https://github.com/pizzaoverhead/FreeIva/assets/7758029/00d7ed49-3531-4fba-9e03-d195a7468ccf)

**Comment by JonnyOThan** [comment]
This is a little harder than it seems at first glance because Restock+ adds a *variant* instead of B9PS.  The attachnodes don't change with the variants, so FreeIva doesn't currently have a way to support making it passable only when that variant is selected.

**Comment by hockeygoalie35** [comment]
Hmm…would it be possible to just support it as is, or is it within the spirit of the mod to stay as accurate as possible?

I’m not super familiar with module manager patches, but is it possible to convert the variant to a B9PS?

**Comment by JonnyOThan** [comment]
Yes, two options that I've considered are applying patches to make the variants use a different set of attachnodes, or switching it to use b9ps instead.

However, those approaches would only work for newly created and launched craft - any old designs or ships in flight wouldn't be updated.

So I think FreeIVA needs to be able to conditionally provide passthrough or an internal based on the part variant.  I think there is a module in WBI tools somewhere that does this already too.

**Comment by JonnyOThan** [comment]
Hmm…there might be a way to tie a b9ps subtype to a stock variant selection…that would probably be the cleanest path forward if it exists.
